stbt_rig.py

```diff
@@ -485,6 +505,236 @@ def cmd_snapshot(args, node):
     node.portal.notify_push()
 
 
+def cmd_setup(args, node_id):
+    this_stbt_rig = os.path.abspath(sys.argv[0])
+    root = find_test_pack_root()
+
+    if os.environ.get("STBT_RIG_SECOND_STAGE") != "1":
+        _, config_parser = read_stbt_conf(root)
+        stbt_version = int(config_parser.get("test_pack", "stbt_version"))
+        python_version = config_parser.get("test_pack", "python_version")
+
+        if not os.path.exists("%s/.venv" % root):
+            python = None
+            for python in (["python"],
+                           ["python%s" % python_version],
+                           ["py", "-%s" % python_version]):
+                try:
+                    o = subprocess.check_output(
+                        python + ["-c", "import sys; print(sys.version)"],
+                        stdin=open(os.devnull)).strip()
+                    if o.startswith(to_bytes("%s." % python_version)):
+                        break
+                except (subprocess.CalledProcessError, OSError):
+                    # Doesn't exist, or there's something wrong with it
+                    pass
+            else:
+                die("Can't find python %s in PATH" % python_version)
+
+            # Create venv
+            subprocess.check_call(python + ['-m', 'venv', '.venv'], cwd=root)
```

Comment:
This only works with Python 3 because 2.7 doesn't have a `venv` module. Is the plan for this `setup` command to be Python3 only? I'm in favour of that but we'd need to make it explicit by checking the stbt.conf's python version and issuing a user-friendly error message.

Reply:
Good spot. Yeah, ok I'll drop Python 2 support from setup and add an error message.  In the unlikely event that there's actually demand I can add it in again with virtualenv support.